### DISCUSSION TITLE
Make Bevy warnings errors?

### DISCUSSION THREAD

**Discussion by 64kramsystem** [fishfight/punchy#139] [Ideas]
Certain warnings that Bevy logs, break the program, but don't panic. I think it'd be useful to cause panics instead, since the app is broken anyway, and if such warnings are accidentally missed by the developer (it happened to me a few times), one risks to waste time.

So, the questions are:

1. what do you think
2. is this even possible?

**Comment by odecay** [comment]
Any examples of this?

**Comment by zicklag** [comment]
> what do you think

I would like to have an example of a warning we'd like to do this on.

> is this even possible?

I believe so. I think we'd setup a custom tracing subscriber that would listen for certain logs and panic when it receives blacklisted logs.

**Comment by 64kramsystem** [comment]
I need to check :smile: 

Today I bashed my head on asset that I've declared in the metadata, but without adding the corresponding file. This was not causing a blank screen, however, I don't think there should be a legitimate case where we declare an asset, but not add the file. Well, at least, not that I'm aware of :smile: 

Regarding the instance of blank screen, I triggered one case while checking a former bug report, but I need to dig a bit.

**Comment by 64kramsystem** [comment]
The case was [this](https://github.com/fishfight/punchy/issues/117). This is a warning:

> 2022-07-21T05:36:21.884767Z  WARN bevy_asset::asset_server: encountered an error while loading an asset: Parser error: The given language subtag is invalid

but it actually causes a black window.

**Reply by zicklag** [reply]
Ah, yeah, that's probably because the game only progresses once the translations finish loading, and in that case, the translations never finish loading because of an error.

We could try to make asset processing errors panic/exit, but we wouldn't be able to do that with hot reload enabled, because then the game would crash as soon as you put a typo into an asset and it failed to load.

Maybe there could be a commandline flag that we pass to enable crashes on asset load failures?

**Reply by 64kramsystem** [reply]
> Maybe there could be a commandline flag that we pass to enable crashes on asset load failures?

Yeah! I think this is a great idea.
